### DISCUSSION TITLE
Something about modcharts

### DISCUSSION THREAD

**Discussion by 3xicFC** [KadeArchive/Kade-Engine#1558] [Ideas]
I think it would be cool for future updates to have modcharts implemented in a trigger like system or as the step events being able to be edited with the debug menu. What do you guys think?

**Comment by isakube** [comment]
converting lua modcharts into song events :flushed:

**Reply by 3xicFC** [reply]
exactly

**Reply by DeceptedPar** [reply]
I have some ideas, it would be more user friendly "obviously"
but you get stuff like;
arrow movement, like moving individual arrows and arrows moving in a snake pattern.
then controlling character opacity, like in foolhardy. you can check a box if you want the change to be smooth or static
screen shake like in tabi - also linking with a ctr effect kind of thing
game windows movement wouldn't be an easy one to do, that might become an external thing people have to do themselves
y'all can also give ideas if y'all want, may help other people out

**Reply by 3xicFC** [reply]
thats about what i was imagining, like not too complicated events, but enough to make good looking modcharts

**Reply by isakube** [reply]
> I have some ideas, it would be more user friendly "obviously"
> but you get stuff like;
> arrow movement, like moving individual arrows and arrows moving in a snake pattern.
> then controlling character opacity, like in foolhardy. you can check a box if you want the change to be smooth or static
> screen shake like in tabi - also linking with a ctr effect kind of thing
> game windows movement wouldn't be an easy one to do, that might become an external thing people have to do themselves
> y'all can also give ideas if y'all want, may help other people out

Only ones possible iirc:

---
Event Name | Arguments
------------ | -------------
Screen Shake | intensity (Float)
Change Dad Character | id (String)
Change BF Character | id (String)

(Will add more later)

---
Theoretically, we could parse arguments with commas, which would literally make everything possible

**Reply by DeceptedPar** [reply]
makes no sense to me but I agree
